From 1ace5fcc687a168fe909939d32b06f6c664d5c12 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Sun, 3 Feb 2019 19:05:20 +0100
Subject: [PATCH 6/7] Clean-up a bit recovery rules

* Re-order some entries
* adbd is always allowed to read adb_keys_file
* Use macros
* vfat is an alias to sdcard_type
* Allow transition to backuptool

Change-Id: Iaecb1552dc83dd3d6c18997d717ac0b02da505e8
---
 common/private/recovery.te | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/common/private/recovery.te b/common/private/recovery.te
index 3eb7a94..6b7ec99 100644
--- a/common/private/recovery.te
+++ b/common/private/recovery.te
@@ -1,12 +1,5 @@
 recovery_only(`
-
-# XXX
-##allow init rootfs:file write;
-get_prop(recovery, sf_lcd_density_prop)
-r_dir_file(recovery, pstorefs)
-
 # Secure adb (setup_adbd)
-allow adbd adb_keys_file:dir search;
 r_dir_file(recovery, adb_keys_file)
 set_prop(recovery, shell_prop)
 
@@ -15,16 +8,15 @@ userdebug_or_eng(`
 allow recovery rootfs:file create_file_perms;
 allow recovery rootfs:file link;
 ')
-allow recovery rootfs:dir { write create rmdir add_name remove_name };
+allow recovery rootfs:dir create_dir_perms;
 
 # Sideload cache
-allow recovery proc_meminfo:file { getattr open read };
+allow recovery proc_meminfo:file r_file_perms;
 
 # Read storage files and directories
 allow recovery tmpfs:dir mounton;
 r_dir_file(recovery, media_rw_data_file)
 r_dir_file(recovery, sdcard_type)
-r_dir_file(recovery, vfat)
 
 # Control properties
 set_prop(recovery, ffs_prop)
@@ -33,7 +25,10 @@ set_prop(recovery, lineage_recovery_prop)
 # Set system properties for various things
 set_prop(recovery, system_prop)
 
-# Sideload backuptool
-allow recovery self:process setexec;
+# Get display density
+get_prop(recovery, sf_lcd_density_prop)
 
+# Switch to ackuptool
+allow recovery self:process setexec;
+domain_trans(recovery, otapreopt_chroot_exec, backuptool)
 ')
-- 
2.17.1

